core/src/commonMain/kotlin/com/ustadmobile/core/controller/ReportEditPresenter.kt

```diff
@@ -0,0 +1,264 @@
+
+        personOneToManyJoinEditHelper.liveList.sendValue(personList)
+        verbDisplaynOneToManyJoinEditHelper.liveList.sendValue(verbList)
+        contentOneToManyJoinEditHelper.liveList.sendValue(contentUidList)
+
+        report.fromDate = DateTime.nowLocal().localEndOfDay.utc.unixMillisLong - 7.days.millisecondsLong
+        report.toDate = DateTime.nowLocal().localEndOfDay.utc.unixMillisLong
+
+        return ReportWithFilters(report, reportFilterList)
+    }
+
+    override fun onLoadFromJson(bundle: Map<String, String>): ReportWithFilters? {
+        super.onLoadFromJson(bundle)
+
+        val entityJsonStr = bundle[ARG_ENTITY_JSON]
+        var editEntity: ReportWithFilters
+        if (entityJsonStr != null) {
+            editEntity = Json.parse(ReportWithFilters.serializer(), entityJsonStr)
+        } else {
+            editEntity = ReportWithFilters()
+        }
+
+        return editEntity
+    }
+
+
+    override fun onSaveInstanceState(savedState: MutableMap<String, String>) {
+        super.onSaveInstanceState(savedState)
+        val entityVal = entity
+        savedState.putEntityAsJson(ARG_ENTITY_JSON, null,
+                entityVal)
+    }
+
+    override fun handleClickSave(entity: ReportWithFilters) {
+
+        if (entity.reportTitle.isNullOrEmpty()) {
+            view.titleErrorText = systemImpl.getString(MessageID.field_required_prompt, context)
+            return
+        } else {
+            view.titleErrorText = null
+        }
+
+
+        GlobalScope.launch(doorMainDispatcher()) {
+
+            if (entity.reportUid != 0L) {
+                repo.reportDao.updateAsync(entity)
+
+                repo.reportFilterDao.insertListAsync(personOneToManyJoinEditHelper.entitiesToInsert)
+                repo.reportFilterDao.updateAsyncList(personOneToManyJoinEditHelper.entitiesToUpdate)
+                repo.reportFilterDao.deactivateByUids(personOneToManyJoinEditHelper.primaryKeysToDeactivate)
+
+                repo.reportFilterDao.insertListAsync(verbDisplaynOneToManyJoinEditHelper.entitiesToInsert)
+                repo.reportFilterDao.updateAsyncList(verbDisplaynOneToManyJoinEditHelper.entitiesToUpdate)
+                repo.reportFilterDao.deactivateByUids(verbDisplaynOneToManyJoinEditHelper.primaryKeysToDeactivate)
+
+                repo.reportFilterDao.insertListAsync(contentOneToManyJoinEditHelper.entitiesToInsert)
+                repo.reportFilterDao.updateAsyncList(contentOneToManyJoinEditHelper.entitiesToUpdate)
+                repo.reportFilterDao.deactivateByUids(contentOneToManyJoinEditHelper.primaryKeysToDeactivate)
+
+                repo.reportDao.updateAsync(entity)
+
+                withContext(doorMainDispatcher()) {
+                    view.finishWithResult(listOf(entity))
+                }
+
+            } else {
+
+                val personUidList = personOneToManyJoinEditHelper.liveList.getValue()?.map { it.entityUid }
```

Comment:
The commitToDatabase function above can also work. Once you have the entity saved to the database, the code should be the same.  The commitToDatabaseFunction can be used to set the foreign key.